platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/CipherUtils.java

```diff
@@ -0,0 +1,205 @@
+import javax.crypto.spec.SecretKeySpec;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.migration.MigrationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CipherUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CipherUtils.class);
+
+  private final Path zipPath;
+  private final Path keyPath;
+  private final Path checksumPath;
+  private final Cipher cipher;
+  private SecretKey secretKey;
+
+  public CipherUtils(Path zipPath) {
+    this(zipPath, getDefaultKeyPathFor(zipPath), getDefaultChecksumPathFor(zipPath));
+  }
+
+  public CipherUtils(Path zipPath, Path keyPath, Path checksumPath) {
+    Validate.notNull(zipPath, "Null zip path");
+    Validate.notNull(keyPath, "Null key path");
+    Validate.notNull(checksumPath, "Null checksum path");
+    this.zipPath = zipPath;
+    this.keyPath = keyPath;
+    this.checksumPath = checksumPath;
+    cipher = createEncryptionCipher();
+  }
+
+  /**
+   * Wraps a given {@link OutputStream} in a {@link CipherOutputStream}
+   *
+   * @param outputStream
+   * @return a {@link CipherOutputStream} for writing encrypted {@link java.util.zip.ZipEntry}
+   *     contents
+   */
+  public CipherOutputStream getCipherOutputStream(OutputStream outputStream) {
+    return new CipherOutputStream(outputStream, cipher);
+  }
+
+  /**
+   * Creates a {@link Cipher} for use during file encryption and stores the secret key in a file
+   *
+   * @throws MigrationException when an invalid key is provided
+   * @throws MigrationException when an invalid key padding is used
+   * @throws MigrationException when and invalid cipher algorithm is used
+   * @return a {@link Cipher} in encrypt mode
+   */
+  private Cipher createEncryptionCipher() {
+    try {
+      Cipher iCipher = Cipher.getInstance(MigrationZipConstants.CIPHER_ALGORITHM);
+      IvParameterSpec ivParameterSpec = new IvParameterSpec(MigrationZipConstants.CIPHER_IV);
+      iCipher.init(Cipher.ENCRYPT_MODE, initKey(keyPath), ivParameterSpec);
+      return iCipher;
+    } catch (InvalidKeyException
+        | NoSuchPaddingException
+        | NoSuchAlgorithmException
+        | InvalidAlgorithmParameterException e) {
+      throw new MigrationException(MigrationZipConstants.KEY_INVALID_ERROR, keyPath, e);
+    }
+  }
+
+  /**
+   * Creates a file containing the checksum for the zip file
+   *
+   * @throws IOException
+   * @throws NoSuchAlgorithmException
+   */
+  public void createZipChecksumFile() throws IOException, NoSuchAlgorithmException {
+    String hash = getChecksum(zipPath);
+    writeStringToFile(checksumPath.toFile(), hash, Charsets.UTF_8);
+  }
+
+  @SuppressWarnings({ //
+    "squid:S2093", /* try-with-resource would require that the input stream is initialized too early */
+    "squid:S2095" /* stream is closed in finally clause */
```

Comment:
❗️ This suppression is no longer needed/correct.

Reply:
Removed